bigquery/google/cloud/bigquery/table.py

```diff
@@ -616,37 +619,50 @@ def _set_properties(self, api_response):
             cleaned['expirationTime'] = float(cleaned['expirationTime'])
         self._properties.update(cleaned)
 
-    def _build_resource(self):
+    def _build_resource(self, fields=[]):
         """Generate a resource for ``create`` or ``update``."""
         resource = {
             'tableReference': {
                 'projectId': self._project,
                 'datasetId': self._dataset_id,
                 'tableId': self.table_id},
         }
-        if self.description is not None:
+        if (
+                self.description is not None and fields == [] or
+                'description' in fields):
             resource['description'] = self.description
 
-        if self.expires is not None:
+        if self.expires is not None and fields == [] or 'expires' in fields:
```

Comment:
What about doing what Client.update_dataset does instead? It seems simpler.

Reply:
I started by doing what Client.update_dataset does (converting from snake case to camel case), but the property names are not consistent due to naming and nesting. For example, expires => expirationTime, partitioning_type => timePartitioning, view_query => view['query'], etc. The table._build_resource() function is very messy though, and @tswast and I discussed how to reorganize the code to make table resource building easier. One thing that might help is creating classes for these properties that can generate their own pieces of the API resource (like Schema and View).

Reply:
I think classes might be overkill. For dataset update I originally had a map from Python settable field names to API field names. Maybe you could have a map from field names to functions that set the field in a resource.

Reply:
A map from field names to functions that populate a resource isn't a bad idea. It would certainly be cleaner than these if statements.

Reply:
Added a map of field names to functions to populate the resource with https://github.com/GoogleCloudPlatform/google-cloud-python/pull/4076/commits/013939afe3eaee6804ac38bd5f693a5d1845e9e1